parsl/executors/base.py

```diff
@@ -87,6 +88,83 @@ def scaling_enabled(self) -> bool:
         """
         pass
 
+    @abstractmethod
+    def status(self) -> Dict[Any, JobStatus]:
+        """Return the status of all jobs/blocks currently known to this executor.
+
+        :return: a dictionary mapping job ids to status strings
+        """
+        pass
+
+    @property
+    @abstractmethod
+    def status_polling_interval(self) -> int:
+        """Returns the interval, in seconds, at which the status method should be called. The
+        assumption here is that, once initialized, an executor's polling interval is fixed.
+        In practice, at least given the current situation, the executor uses a single task provider
+        and this method is a delegate to the corresponding method in the provider.
+
+        :return: the number of seconds to wait between calls to status() or zero if no polling
+        should be done
+        """
+        pass
+
+    @property
+    @abstractmethod
+    def error_management_enabled(self) -> bool:
```

Comment:
A different structure for this is to make a new ABC that defines all of the error management abstract methods; classes which implement error management inherit from that, classes which do not, do not. That removes the need for NoStatusMixin and a bunch of (I think?) dummy methods there that shouldn't be called but still need implementing.

Reply:
True. Feel free... :)

Reply:
With this we'd need to touch every executor. It would be much more easier to start with just htex supporting the full status reporting feature set.

Reply:
The easiest is to start with what's there, which is this PR. Going back and changing all the executors that needed to be touched and untouching them (as well as figuring out a clean way for untouched executors to coexist with touched executors) is, as far as I'm concerned, quite clearly not easier.